### PR TITLE
Created Data model

### PR DESCRIPTION
**Pull request by NalianAchei2020**
#  Acomplishment
- Connected to PostgreSQL database
- Created and run the necessary migration files.
- Table and column names match the ERD diagram.
- All columns that are foreign keys  have a corresponding index

### CODE REVIEW

**Review by julie-ify — COMMENTED** [review]
Hi @NalianAchei2020,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* Your migrations are failing. You did not set up your migration files properly. You should create users table, followed by posts, then comments and likes. But since you created comments and likes before posts they raise errors for a new user setting up your app because comments and likes tables depend on posts table, which chronologically is not created until later.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by NalianAchei2020** [comment]
> Hi @NalianAchei2020,
> 
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> 
> ## Invalid Code Review Request
> * Your migrations are failing. You did not set up your migration files properly. You should create users table, followed by posts, then comments and likes. But since you created comments and likes before posts they raise errors for a new user setting up your app because comments and likes tables depend on posts table, which chronologically is not created until later.
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by NalianAchei2020** [comment]
Hi @julie-ify 
From my commit history, I created it in the order you just mentioned. The post was created before comments and likes.

File: db/migrate/20240111192555_create_posts.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 class CreatePosts < ActiveRecord::Migration[7.1]
   def change
     create_table :posts do |t|
-      t.references :author, null: false, foreign_key: true
+      t.references :user, null: false, foreign_key: true
```
Comment (mckent05) [inline db/migrate/20240111192555_create_posts.rb]: - You have done well so far, but as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png), the column in the post table that references the user table should be named `author`. Please fix this 🙏.

```ruby
     t.references :author, foreign_key: { to_table: 'users' }
```

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @NalianAchei2020, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job creating migration files to setup your database tables 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20240111192254_create_users.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@ def change
       t.string :name
       t.string :photo
       t.text :bio
-      t.datetime :updated_at
-      t.datetime :created_at
+      t.datetime :updatedAt
+      t.datetime :createdAt
```
Comment (TedLivist) [inline db/migrate/20240111192254_create_users.rb:8]: - [ ] **[OPTIONAL]** I would suggest you remove these columns. Besides, the fact they do not follow snake case rule for table columns, there is no use for these `updateAt` and `createdAt` columns. The `t.timestamps` takes care of the `updated_at` and `created_at` columns that we need.

File: db/migrate/20240111192555_create_posts.rb [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
   def change
     create_table :posts do |t|
-      t.references :author, null: false, foreign_key: true
+      t.references :author, null: false, foreign_key: { to_table: 'users' }
       t.string :title
       t.text :text
-      t.datetime :created_at
-      t.datetime :updated_at
+      t.datetime :createdAt
+      t.datetime :updatedAt
```
Comment (TedLivist) [inline db/migrate/20240111192555_create_posts.rb:8]: - [ ] **[OPTIONAL]** I would suggest you remove these columns. Besides, the fact they do not follow snake case rule for table columns, there is no use for these `updateAt` and `createdAt` columns. The `t.timestamps` takes care of the `updated_at` and `created_at` columns that we need.

File: db/migrate/20240111192846_create_likes.rb [hunk 4]
```diff
@@ -3,8 +3,8 @@ def change
     create_table :likes do |t|
       t.references :user, null: false, foreign_key: true
       t.references :post, null: false, foreign_key: true
-      t.datetime :created_at
-      t.datetime :updated_at
+      t.datetime :createdAt
+      t.datetime :updatedAt
```
Comment (TedLivist) [inline db/migrate/20240111192846_create_likes.rb:7]: - [ ] **[OPTIONAL]** I would suggest you remove these columns. Besides, the fact they do not follow snake case rule for table columns, there is no use for these `updateAt` and `createdAt` columns. The `t.timestamps` takes care of the `updated_at` and `created_at` columns that we need.

File: db/migrate/20240111193049_create_comments.rb [hunk 5]
```diff
@@ -4,8 +4,8 @@ def change
       t.references :user, null: false, foreign_key: true
       t.references :post, null: false, foreign_key: true
       t.text :text
-      t.datetime :updated_at
-      t.datetime :created_at
+      t.datetime :updatedAt
+      t.datetime :createdAt
```
Comment (TedLivist) [inline db/migrate/20240111193049_create_comments.rb:8]: - [ ] **[OPTIONAL]** I would suggest you remove these columns. Besides, the fact they do not follow snake case rule for table columns, there is no use for these `updateAt` and `createdAt` columns. The `t.timestamps` takes care of the `updated_at` and `created_at` columns that we need.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @NalianAchei2020,

✔️ Use of PostgreSQL for database

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**